File: CS/Sample/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using Sample.Models;

namespace Sample.Controllers {
    public class HomeController : Controller {
        PersonsList list = new PersonsList();

        public ActionResult Index() {
            return View(list.GetPersons());
        }

        public ActionResult GridViewTemplatePartial() {
            return PartialView(list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingAddNew([ModelBinder(typeof(DevExpressEditorsBinder))] Person person) {
            if (ModelState.IsValid)
                list.AddPerson(person);
            return PartialView("GridViewTemplatePartial", list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingUpdate([ModelBinder(typeof(DevExpressEditorsBinder))] Person personInfo) {
            if (ModelState.IsValid)
                list.UpdatePerson(personInfo);
            return PartialView("GridViewTemplatePartial", list.GetPersons());
        }

        public ActionResult EditingDelete(int personId) {
            list.DeletePerson(personId);
            return PartialView("GridViewTemplatePartial", list.GetPersons());
        }
    }
}

File: VB/Sample/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports Sample.Models

Namespace Sample.Controllers

    Public Class HomeController
        Inherits Controller

        Private list As PersonsList = New PersonsList()

        Public Function Index() As ActionResult
            Return View(list.GetPersons())
        End Function

        Public Function GridViewTemplatePartial() As ActionResult
            Return PartialView(list.GetPersons())
        End Function

        <HttpPost, ValidateInput(False)>
        Public Function EditingAddNew(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal person As Person) As ActionResult
            If ModelState.IsValid Then list.AddPerson(person)
            Return PartialView("GridViewTemplatePartial", list.GetPersons())
        End Function

        <HttpPost, ValidateInput(False)>
        Public Function EditingUpdate(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal personInfo As Person) As ActionResult
            If ModelState.IsValid Then list.UpdatePerson(personInfo)
            Return PartialView("GridViewTemplatePartial", list.GetPersons())
        End Function

        Public Function EditingDelete(ByVal personId As Integer) As ActionResult
            list.DeletePerson(personId)
            Return PartialView("GridViewTemplatePartial", list.GetPersons())
        End Function
    End Class
End Namespace
